app/components/Admin/Room/RoomDetail.jsx

```diff
@@ -1,20 +1,68 @@
 "use strict"
+import R from 'ramda';
 import dom from '../../../utils/dom';
 import Room from './room-class.js'
+import Field from '../../Form/Field-class';
+import { dispatch, dispatchAsync } from '../../../index'
 
 var tempRoom = new Room();
 
-
+const {is, isEmpty} = R;
+
+const isNonEmptyString = (value) => is(String, value) && !isEmpty(value)
+const isNotNegative = (value) => +value >= 0;
+const isPositive = (value) => +value > 0;
+
```

Comment:
I like these validation lambda functions `isNonEmptyString`, `isNotNegative` and `isPositive`